### DISCUSSION TITLE
Proposal: Promote JSON Schema consultancy/contracting of TSC and Ambassadors

### DISCUSSION THREAD

**Discussion by jviotti** [json-schema-org/community#779] [Governance and Policy]
While the JSON Schema team and key members offer some degree of "free" help to the rest of the JSON Schema community, some TSC members (including myself) came across companies seeking and willing to pay for additional hands-on help ranging from laying out their JSON Schema strategy on their products, building custom JSON Schema tooling, reviews of their schema collections, training, and more. The problem is that these ready-to-pay companies often don't know who to reach out to for consultancy, and they might not even realise this is an option. I'd like to fix this while funding the JSON Schema organisation as a by-product.

**Proposal**

- Members of either the TSC or the Ambassadors program can opt-in to be listed as "official" consultants/contractors. This has the added bonus of making the Ambassadors program more sought after. Of course, any member can opt-out at any given point (i.e. if they get full-time employment that prevents them from engaging in consulting)
- The JSON Schema website will have a new page where it lists the people open to consultancy/contracting in some fair ordering (alphabetic?), some information about them, and how to reach out. The rate and payment method is totally up to the member and the company. The JSON Schema organisation has no say on this
 
- On the interest of the org, if work requires more capacity than what a single member can produce or a member cannot take the job, the members should prefer bringing on board other TSC or Ambassadors instead of external people, so we can all help each other getting work 
- To make it a bit more appealing to companies, it'd be interesting to make it so that if a company hires any of us through the website, their logo gets showcased on the JSON Schema website under a certain category, like "friends of JSON Schema"?

**Roadmap**

Assuming this is agreed upon, the action items are as follows:

- [x] Add a JSON file to the `community` repo that includes people that want to take part of this, same as we currently do with ambassadors (https://github.com/json-schema-org/community/tree/main/programs/ambassadors)
    - PR: https://github.com/json-schema-org/community/pull/786
- [x] Render a page on the JSON Schema website with this content at json-schema.org/services or whatever
    - PR: https://github.com/json-schema-org/website/pull/1015
- [x] Link to this page from other relevant pages of the website
    - PR: https://github.com/json-schema-org/website/pull/1015
- [ ] Announce this new program on social media and/or blog post

**Comment by jdesrosiers** [comment]
I suggest that we make it a suggested donation rather than a required fee. We have no legal standing to demand a commission because we aren't a legal entity of any kind. I don't know if others are the same, but I tend to feel better about being asked to make a donation than being asked to pay a fee.

**Reply by gregsdennis** [reply]
I think this is more about companies looking to get official support directly from org members.  We want to promote that companies support our members and, in turn, the project.

We, as individuals, do have the right to be paid for our labor.

**Reply by jviotti** [reply]
Another alternative is not paying/donating a percentage back to OpenCollective, under the assumption that hiring org members already implicitly funds JSON Schema by funding the time of the org members. i.e. we pay back to JSON Schema with our time rather than with a cut of our gig.

**Reply by jviotti** [reply]
As discussed in today's call, I completely removed the commission idea from the proposal. Unless there is any other comment, I'll try to make some proper progress on it this week.

**Comment by jviotti** [comment]
Here goes the first PR: https://github.com/json-schema-org/community/pull/786. I tried to copy the Ambassador template as much as possible. Any feedback much appreciated 🙏🏻

**Comment by jviotti** [comment]
@benjagm Can you give me a bit of input on how to integrate this in on the website? We have the contractors JSON file on the `community` repo (https://github.com/json-schema-org/community/blob/main/programs/contractors/contractors.json) but its unclear to me how to feed that into the website for rendering. 

I copied the approach based on the ambassadors list, but then realised that's not integrated into the website yet. I guess we want those files to be fed into the website similar to the `data` directory right now? (https://github.com/jviotti/json-schema-website/tree/main/data)

How were you planning to do it, so I attempt to do it in the same way? Git submodules? Will ambassadors get rendered soon? If so, maybe we can just wait a bit.

**Comment by jviotti** [comment]
Hi everybody! We have a reviewable PR rendering this into the website here: https://github.com/json-schema-org/website/pull/1015. Can you please take a look?

**Comment by jviotti** [comment]
@benjagm Now that we have the website up and running, let's discuss a bit more how we want to announce this? How would you want to do it? Social media? Blog post?

Also cc @jdesrosiers as you shared interest in joining the program. It would be great if you add your profile there too so I'm not the only one!

**Reply by gregsdennis** [reply]
I think both a blog post and socials.  We (TSC & ambassadors, whoever feels comfortable) should promote and push it.
